src/services/token/token.js

```diff
@@ -5,21 +5,47 @@ import { getLogger } from "../../utils/logger";
 
```

Comment:
TODO: let's refactor this to be encapsulated so that we don't introduce bugs down the line